### PR TITLE
Add SNI client support

### PR DESCRIPTION
**Pull request by Adam-**

### CODE REVIEW

File: src/Socket.cpp [hunk 1]
```diff
@@ -126,6 +126,12 @@ void CZNCSock::SSLHandShakeFinished() {
 	CallSockError(errnoBadSSLCert, sErrorMsg);
 	Close();
 }
+
+bool CZNCSock::SNIConfigureClient(CS_STRING & sHostname)
```
Comment (DarthGandalf) [inline src/Socket.cpp]: It's `CString&` in ZNC.
Also please follow the style around.

**Comment by DarthGandalf** [comment]
#948 

What happens if `m_HostToVerifySSL` is IP instead of hostname?

**Comment by Adam-** [comment]
> It's CString& in ZNC.

Several of the related declarations in CZNCSock use the CS_STRING & format but sure I'll change it.

> What happens if m_HostToVerifySSL is IP instead of hostname?

That doesn't matter, the server name sent would just be the IP.

**Comment by DarthGandalf** [comment]
> Several of the related declarations in CZNCSock use the CS_STRING & format but sure I'll change it.

I meant not just the space, but CString vs CS_STRING, and placement of {
The only other instance of CS_STRING should be fixed though, thanks for pointing it out.

**Pull request closed without merging**